File: stepped_cable_lengths.m
function [stepped_cables] = stepped_cable_lengths(ideal_cable_length, step_length)
stepped_cables = zeros(8,length(ideal_cable_length));
for i = 1:8
    stepped_cables(i,:) = ceil((ideal_cable_length(i,:) - ideal_cable_length(i,1))./step_length).*step_length + ideal_cable_length(i,1);
end

end

File: Motion_Generation_Python/stepped_cable_lengths.py
def stepped_cable_lengths(ideal_cable_length, step_length):
    
    import numpy as np

    stepped_cables = np.zeros([8, len(ideal_cable_length[0])]);

    
    for i in range (8):
        #stepped_cables[i,:] = np.ceil((ideal_cable_length[i,:] - ideal_cable_length[i,0])/step_length)*step_length + ideal_cable_length[i,0];
        stepped_cables[i,:] = np.ceil((ideal_cable_length[i,:])/step_length)*step_length
    
    print('stepped cable lengths', stepped_cables)
    print('stepped cable length array shape', stepped_cables.shape)
    return stepped_cables